legacy/src/main/java/org/pentaho/amazon/AmazonS3FileSystemBootstrap.java

```diff
@@ -42,28 +46,37 @@
  */
 @KettleLifecyclePlugin( id = "AmazonS3FileSystemBootstrap", name = "Amazon S3 FileSystem Bootstrap" )
 public class AmazonS3FileSystemBootstrap implements KettleLifecycleListener {
-  private static Class<?> PKG = AmazonS3FileSystemBootstrap.class;
+  private static Class<?> amazonS3FileSystemBootstrapClass = AmazonS3FileSystemBootstrap.class;
   private LogChannelInterface log = new LogChannel( AmazonS3FileSystemBootstrap.class.getName() );
+  private Supplier<ConnectionManager> connectionManager = ConnectionManager::getInstance;
 
   /**
    * @return the i18n display text for the S3 file system
    */
   public static String getS3FileSystemDisplayText() {
-    return BaseMessages.getString( PKG, "S3VfsFileChooserDialog.FileSystemChoice.S3.Label" );
+    return BaseMessages
+      .getString( amazonS3FileSystemBootstrapClass, "S3VfsFileChooserDialog.FileSystemChoice.S3.Label" );
   }
 
   @Override
   public void onEnvironmentInit() throws LifecycleException {
     try {
       // Register S3 as a file system type with VFS
       FileSystemManager fsm = KettleVFS.getInstance().getFileSystemManager();
-      if ( fsm instanceof DefaultFileSystemManager ) {
-        if ( !Arrays.asList( fsm.getSchemes() ).contains( S3FileProvider.SCHEME ) ) {
-          ( (DefaultFileSystemManager) fsm ).addProvider( S3FileProvider.SCHEME, new S3FileProvider() );
-        }
+      if ( fsm instanceof DefaultFileSystemManager && !Arrays.asList( fsm.getSchemes() )
+        .contains( S3FileProvider.SCHEME ) ) {
+        ( (DefaultFileSystemManager) fsm ).addProvider( S3FileProvider.SCHEME, new S3FileProvider() );
       }
+
+      if ( connectionManager.get() != null ) {
+        connectionManager.get().addConnectionProvider( S3FileProvider.SCHEME, new S3Provider() );
+      }
+
+      VFSLookupFilter vfsLookupFilter = new VFSLookupFilter();
```

Comment:
I feel like all of this should be in the null guard on line 71

Reply:
Cut and pasted. I agree, let me update.

Reply:
I agree I don't think we want to get the lookup filter if the connectionManager is null.